### DISCUSSION TITLE
Using extra parameters for Azure API connections

### DISCUSSION THREAD

**Discussion by vswraith** [microsoft/autogen#2338] [Q&A]
Is it possible to add extra openai parameters to connect to Azure endpoints?
 Eg- our endpoint is deployed at 
https://<url>.azure-api.net/deployments/gpt-4/chat/completions?api-version=2023-05-01.

But by default, the code adds /openai to it.
Hence Autogen hits at 
https://<url>.azure-api.net/openai/deployments/gpt-4/chat/completions?api-version=2023-05-01 and this does not work. 
In python, using the openAI lib, there is a way to override this by setting-

```
openai.ChatCompletion.azure_api_prefix=''
openai.Completion.azure_api_prefix=''
```
Can we do something similar for Autogen?

**Comment by victordibia** [comment]
Hi @vswraith ,

Thanks for mentioning this issue. 
Can you give a brief example of how you are specifying your OAI_CONFIG?

**Comment by sonichi** [comment]
The same lines should work. Alternatively you can also pass them in the `.create()` method of `autogen.Completion` etc.

**Comment by gregnwosu** [comment]
how do you do this? please show me an example i need to set cache_prompt to false from initiate chat for ollama
